tests/ut/test_decorators.py

```diff
@@ -197,6 +198,16 @@ class Dummy:
             assert memory_mock_cache.get.call_count == 0
             assert memory_mock_cache.set.call_count == 0
 
+    @pytest.mark.asyncio
+    async def test_cached_keeps_signature(self):
+        @cached()
+        async def what(self, a, b):
+            return "1"
+
+        assert what.__name__ == "what"
+        assert str(inspect.signature(what)) == '(self, a, b)'
+        assert inspect.getfullargspec(what.__wrapped__).args == ['self', 'a', 'b']
```

Comment:
yeah, my point was more that if Zope uses `inspect.signature` then it will work. If it uses `inspect.getfullargspec` it won't because the real signature is under the `__wrapped__` attribute of the function and `getfullargspec` doesn't use it (while `signature` does :))